src/main/java/org/reaktivity/nukleus/kafka/internal/stream/TopicMessageDispatcher.java

```diff
@@ -38,15 +39,16 @@
     private final boolean compacted;
 
     protected TopicMessageDispatcher(
-        PartitionIndex[] indexes,
+        TopicCache cache,
+        int partitionCount,
         boolean compacted,
```

Comment:
Hoist `partitionCount` and `compacted` into `TopicCache` interface (as methods)?

Reply:
Done for compacted. I don't want to do it for partitionCount because it would preclude using a single instance of DefaultTopicCache (now StreamingTopicCache), the empty cache implementation.